test/src/jupyterhub_config/custom_spawner_test.py

```diff
@@ -1 +1,103 @@
-from jupyterhub_config.custom_spawner import *
\ No newline at end of file
+import logging
+import subprocess
+import unittest
+from subprocess import CalledProcessError
+from unittest.mock import patch, MagicMock
+
+import pytest
+
+from jupyterhub_config.custom_spawner import VirtualEnvSpawner
+
+
+@pytest.fixture
+def mock_spawner():
+    spawner = VirtualEnvSpawner()
+    return spawner
```

Comment:
Seems like you could just get rid of this fixture and create it in the test methods, since there's no mocks any more

Reply:
oh yea. removed spawner fixture.